keras_autodoc/get_signatures.py

```diff
@@ -1,5 +1,4 @@
 import warnings
-from sphinx.util.inspect import Signature
```

Comment:
The advantage of using sphinx to get the signature (types hints too) compare to plain signature from `inspect` is that there are some changes to make it easier to read. I tried your pull request on the autokeras documentation, it looks great, but the types are slightly more difficult to read now:

`Optional[int]` becomes `typing.Union[int, NoneType]`

One solution may be to continue to use `Signature` from sphinx. This also makes sure we're futureproof concerning type hints. Since it's a private API of sphinx, there is no documentation, but here is the source code:  https://github.com/sphinx-doc/sphinx/blob/3.x/sphinx/util/inspect.py#L680  It might be possible to just use the function that formats the type hints. You can do what you think is best here.

Reply:
Update: I think it'd more robust/future-proof to rely on Python's built-in typing support instead of using that private sphinx API. According to the Python docs, [`typing.get_type_hints()`](https://docs.python.org/3/library/typing.html#typing.get_type_hints) should enable that for us, but it doesn't seem to work for me. I'm going to [give my SO question on this](https://stackoverflow.com/questions/61320106/python-typing-get-type-hints) a few days to hopefully get a nice built-in answer, and then decide how to proceed. (I'll wait to work on your other comments until then too.)

Reply:
If we can achieve a similar result with the public API, I'm all for it.

Reply:
Is it just the `Optional[x]` becoming `typing.Union[x, NoneType]` now that get less readable, or other things as well? It looks like [`Optional` is just syntactic sugar](https://docs.python.org/3/library/typing.html#typing.Optional) and internally gets saved as a `Union`, but if that's the only problem I think just treating that as a special case and substituting it manually is still a bit nicer than using the private sphinx API.

Reply:
Manual substitution may not be trivial. 

There are three differences:
* Optional
* `typing.` is removed
* NoneType is displayed instead of None.

If we can acheive that with only the python public api it would be cool indeed.

Reply:
Yeah you're right; after looking at the complications in [`sphinx.util.typing.stringify()`](https://github.com/sphinx-doc/sphinx/blob/3.x/sphinx/util/typing.py#L42) I think it's probably not worth the maintenance burden to implement this ourselves. I'm now just calling that (9b73508).